pulsar/client_impl.go

```diff
@@ -168,12 +168,26 @@ func (c *client) Subscribe(options ConsumerOptions) (Consumer, error) {
 }
 
 func (c *client) CreateReader(options ReaderOptions) (Reader, error) {
-	reader, err := newReader(c, options)
+	topics, err := c.TopicPartitions(options.Topic)
 	if err != nil {
 		return nil, err
 	}
-	c.handlers.Add(reader)
-	return reader, nil
+	if len(topics) <= 1 {
+		reader, err := newReader(c, options)
```

Comment:
What's the different between a reader and a multi reader? It seems like they should be the same but just consume a different number of partitions?